lib/src/util/withClass.tsx

```diff
@@ -4,16 +4,30 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import React, { FunctionComponent } from 'react';
+import React, { forwardRef, DetailedHTMLProps, SVGProps } from 'react';
 import classnames from 'classnames';
 
+type NativeElTypeFromDetailedHTMLProps<E extends keyof JSX.IntrinsicElements> =
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  JSX.IntrinsicElements[E] extends DetailedHTMLProps<any, infer A> ? A : never;
+
+type NativeElTypeFromSVGProps<E extends keyof JSX.IntrinsicElements> =
+  JSX.IntrinsicElements[E] extends SVGProps<infer A> ? A : never;
+
+type NativeElType<E extends keyof JSX.IntrinsicElements> =
+  NativeElTypeFromDetailedHTMLProps<E> extends never
+    ? NativeElTypeFromSVGProps<E> : NativeElTypeFromDetailedHTMLProps<E>;
```

Comment:
Was it necessary to have it be three types like this? I would have expected you to check against both `DetailedHTMLProps` and `SVGProps` in a single nested ternary, which would probably be more straightforward